File: src/main/java/com/paragon/client/managers/social/SocialManager.java
package com.paragon.client.managers.social;

import com.paragon.Paragon;

import java.util.ArrayList;
import java.util.List;

/**
 * @author Wolfsurge
 */
public class SocialManager {

    public final List<Player> players = new ArrayList<>();

    public SocialManager() {
        Paragon.INSTANCE.getLogger().info("Loaded Social Manager");
    }

    /**
     * Checks if a given player name is a friend
     *
     * @param name The name to check
     * @return If the given player is a friend
     */
    public boolean isFriend(String name) {
        for (Player player : players) {
            if (player.getName().equalsIgnoreCase(name) && player.getRelationship() == Relationship.FRIEND) {
                return true;
            }
        }

        return false;
    }

    /**
     * Checks if a given player name is neutral
     *
     * @param name The name to check
     * @return If the given player is neutral
     */
    public boolean isNeutral(String name) {
        for (Player player : players) {
            if (player.getName().equalsIgnoreCase(name) && player.getRelationship() == Relationship.NEUTRAL) {
                return true;
            }
        }

        return false;
    }

    /**
     * Checks if a given player name is an enemy
     *
     * @param name The name to check
     * @return If the given player is an enemy
     */
    public boolean isEnemy(String name) {
        for (Player player : players) {
            if (player.getName().equalsIgnoreCase(name) && player.getRelationship() == Relationship.ENEMY) {
                return true;
            }
        }

        return false;
    }

    /**
     * Adds a player to our players list
     *
     * @param player The player to add
     */
    public void addPlayer(Player player) {
        players.removeIf(player1 -> player1.getName().equalsIgnoreCase(player.getName()));

        players.add(player);
    }

    public void removePlayer(String name) {
        players.removeIf(player -> player.getName().equalsIgnoreCase(name));
    }

}

File: src/main/java/com/paragon/client/managers/social/SocialManager.kt
package com.paragon.client.managers.social

import com.paragon.Paragon

/**
 * @author SooStrator1136
 */
class SocialManager {

    val players: MutableList<Player> = ArrayList()

    init {
        Paragon.INSTANCE.logger.info("Loaded Social Manager")
    }

    /**
     * Checks if a given player name is a friend
     *
     * @param name The name to check
     * @return If the given player is a friend
     */
    fun isFriend(name: String): Boolean {
        for (player in players) {
            if (player.relationship == Relationship.FRIEND && player.name.equals(name, true)) {
                return true
            }
        }
        return false
    }

    /**
     * Checks if a given player name is neutral
     *
     * @param name The name to check
     * @return If the given player is neutral
     */
    fun isNeutral(name: String): Boolean {
        for (player in players) {
            if (player.relationship == Relationship.NEUTRAL && player.name.equals(name, true)) {
                return true
            }
        }
        return false
    }

    /**
     * Checks if a given player name is an enemy
     *
     * @param name The name to check
     * @return If the given player is an enemy
     */
    fun isEnemy(name: String): Boolean {
        for (player in players) {
            if (player.relationship == Relationship.ENEMY && player.name.equals(name, true)) {
                return true
            }
        }
        return false
    }

    /**
     * Adds a player to our players list
     *
     * @param player The player to add
     */
    fun addPlayer(player: Player) {
        players.removeIf { player1: Player -> player1.name.equals(player.name, true) }
        players.add(player)
    }

    /**
     * Removes a player from our players list
     *
     * @param name The name of the player to remove
     */
    fun removePlayer(name: String) {
        players.removeIf { player: Player -> player.name.equals(name, true) }
    }

}